### PR TITLE
Add more loggings for GradientBuilder

### PR DESCRIPTION
**Pull request by SherlockNoMad**
**Description**: As title

### CODE REVIEW

File: orttraining/orttraining/core/graph/gradient_builder.cc [hunk 1]
```diff
@@ -1234,16 +1234,17 @@ IMPLEMENT_GRADIENT_BUILDER(GetGeluGradient) {
 namespace {
 std::vector<NodeDef> GetBiasGeluGradNodes(
     bool use_approximation,
-    const ArgDef& dY, const ArgDef& X, const ArgDef& B,                    // inputs
-    const ArgDef& dX, const ArgDef& dB,                                    // outputs
-    const ArgDef& b_axes, const ArgDef& b_shape, const ArgDef& x_shape) {  //intermediate args
+    const ArgDef& dY, const ArgDef& X, const ArgDef& B,  // inputs
+    const ArgDef& dX, const ArgDef& dB,                  // outputs
+    const ArgDef& b_axes, const ArgDef& b_shape, const ArgDef& x_shape,
+    const std::string& node_name) {  //intermediate args
```
Comment (edgchen1) [inline orttraining/orttraining/core/graph/gradient_builder.cc:1240]: nit: this comment is not lined up correctly anymore

File: orttraining/orttraining/core/graph/gradient_builder_base.cc [hunk 2]
```diff
@@ -9,11 +9,29 @@
...
+  output << "[";
+  for (auto& dim : dims) {
+    if (dim.has_dim_value()) {
+      output << dim.dim_value() << ",";
+    }
+    if (dim.has_dim_param()) {
+      output << dim.dim_param() << ",";
+    }
+  }
+  output.seekp(-1, output.cur);
```
Comment (edgchen1) [inline orttraining/orttraining/core/graph/gradient_builder_base.cc:23]: what if dims is empty?

File: orttraining/orttraining/core/graph/gradient_builder_base.h [hunk 3]
```diff
@@ -244,6 +245,8 @@ class GradientBuilderBase {
                                  const ArgDef& reduce_axes,
                                  std::vector<NodeDef>& output) const;
 
+  std::string NodeName() const { return node_->Name(); }
```
Comment (edgchen1) [inline orttraining/orttraining/core/graph/gradient_builder_base.h:248]: nit: can we return `const string&`?

**Review by Jeff-GiJoe — APPROVED** [review]

**Review by edgchen1 — DISMISSED** [review]

**Review by edgchen1 — APPROVED** [review]

**Pull request merged**